runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java

```diff
@@ -190,6 +186,23 @@ public LogData(DataType type, final Object object) {
         }
     }
 
+    /**
+     * Assign a given token to this log data.
+     *
+     * @param token the token to use
+     */
+    @Override
+    public void useToken(IToken token) {
+        setGlobalAddress(token.getSequence());
+        setEpoch(token.getEpoch());
+        if (token.getBackpointerMap().size() > 0) {
+            setBackpointerMap(token.getBackpointerMap());
+        }
+        if (payload.get() instanceof LogEntry) {
+            ((LogEntry) payload.get()).setGlobalAddress(token.getSequence());
```

Comment:
Oh, I see, this will fix the issue i mentioned above.

Reply:
Yes you're right.

Reply:
Why useToken is not ILogData default method anymore？

Reply:
Because I need to access `payload`, but `payload` is not in `ILogData` interface.